scripts/install_ebpf.psm1

```diff
@@ -219,48 +146,171 @@ function Install-eBPFComponents
+    $process = Start-Process -FilePath $VcRedistPath -ArgumentList "/quiet", "/norestart" -Wait -PassThru
+    if ($process.ExitCode -ne 0) {
+        Write-Log("Visual C++ Redistributable installation FAILED. Exit code: $($process.ExitCode)") -ForegroundColor Red
+        throw ("Visual C++ Redistributable installation FAILED. Exit code: $($process.ExitCode)")
+    }
+    Write-Log("Cleaning up...")
+    Remove-Item $VcRedistPath -Force
+    Write-Log("Visual C++ Redistributable installation completed successfully!") -ForegroundColor Green
+
+    # Copy the VC debug runtime DLLs to the system32 directory,
+    # so that debug versions of the MSI can be installed (i.e. export_program_info.exe will not fail).
+    Write-Log("Copying VC debug runtime DLLs to the $system32Path directory...")
+    # Test is the VC debug runtime DLLs are present in the working directory (indicating a debug build).
+    $VCDebugRuntime = $VCDebugRuntime | Where-Object { Test-Path (Join-Path $WorkingDirectory $_) }
+    if (-not $VCDebugRuntime) {
+        Write-Log("VC debug runtime DLLs not found in the working directory (i.e. release build). Skipping this step.") -ForegroundColor Yellow
+    } else {
+        $system32Path = Join-Path $env:SystemRoot "System32"
+        $VCDebugRuntime | ForEach-Object {
+            $sourcePath = Join-Path $WorkingDirectory $_
+            $destinationPath = Join-Path $system32Path $_
+            Write-Log("Copying '$sourcePath' to '$destinationPath'...")
+            Copy-Item -Path $sourcePath -Destination $destinationPath -Force
+        }
+        Write-Log("VC debug runtime DLLs copied successfully!") -ForegroundColor Green
+    }
+
+    # Install the MSI package.
+    $arguments = "/i $MsiPath ADDLOCAL=ALL /qn /norestart /l*v msi-install.log"
+    Write-Log("Installing the eBPF MSI package: 'msiexec.exe $arguments'...")
+    $process = Start-Process -FilePath msiexec.exe -ArgumentList $arguments -Wait -PassThru
+    if ($process.ExitCode -ne 0) {
+        Write-Log("MSI installation FAILED. Exit code: $($process.ExitCode)") -ForegroundColor Red
+
+        # For clear readability within the CICD pipeline and final uploaded log output,
+        # read each line of the log file and print it (otherwise all the log content is printed as a single line).
+        Write-Log("Contents of msi-install.log:")
+        Get-Content -Path "msi-install.log" | ForEach-Object {
+            Write-Log($_)
+        }
+        throw ("MSI installation FAILED. Exit code: $($process.ExitCode)")
     }
-    if (Test-Path -Path "testing\testing") {
-        Copy-Item testing\testing\*.sys -Destination "$Env:systemroot\system32\drivers" -Force -ErrorAction Stop 2>&1 | Write-Log
+    Write-Log("eBPF MSI installation completed successfully!") -ForegroundColor Green
+
+    # Install the extra drivers that are not installed by the MSI package.
+    $EbpfDrivers.GetEnumerator() | ForEach-Object {
```

Comment:
why is ebpfcore not installed by the msi?

Reply:
it is... not sure why you ask that

Reply:
Because the variable $EbpfDrivers contains the string ebpfcore, and the comment right above the call to Getenumerator says "extra drivers that are not installed by the MSI package". Am I reading this wrong?

Reply:
yes, you're missing the logic implemented in the loop...